### PR TITLE
resolver result holds versionless and platform errors

### PR DESCRIPTION
**Pull request by rsherget**
Feature Resolver result will now keep track of some versionless-specific information and errors such as missing platforms, resolved platforms, and what versionless features resolve to.

### CODE REVIEW

**Comment by rsherget** [comment]
#build

File: dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureResolverImpl.java [hunk 1]
```diff
@@ -320,13 +320,75 @@ private List<String> collectEnvironmentPlatforms(Repository repo, Collection<Str
...
             } else {
-                Tr.error((TraceComponent) tc, "UNKNOWN_PLATFORM_VALUE_ENV_VAR", plat);
+                selectionContext.getResult().addMissingPlatform(plat);
             }
         }
 
         return compatibilityFeatures;
     }
 
+    /**
```
Comment (cbridgha) [inline dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureResolverImpl.java:330]: This looks ok, but I'll need to test the install scenario where no platforms exist in the kernel package, and install is trying to add jakartaeefeatures, so the resolver would need to find these platforms in the remote repo - I think this should still work ok - but be careful caching this value.

**Review by cbridgha — APPROVED** [review]
If you can add a few comments what these methods are doing?    I added one worry regarding remote repositories, but I'll be testing that today...  In any case I approve

**Comment by cbridgha** [comment]
Just questions...  Now that we have both  getResolvedPlatforms() and  getMissingPlatforms() - I assume getMissingPlatforms() will only report on misspelled, or platforms not found in the current repo?  Then the case where no platforms are defined, and a platform can't be determined...  getResolvedPlatforms() will be empty?

**Comment by rsherget** [comment]
Correct, missingPlatforms will contain the specified platforms that are misspelled or non-existent, resolvedPlatforms will contain what is specified. If nothing is specified/nothing can be determined, then it will be empty

**Pull request merged**